From 1634249657d8c838ca22eb44024d851df8bcc081 Mon Sep 17 00:00:00 2001
From: Olivier Moysan <olivier.moysan@foss.st.com>
Date: Wed, 2 Oct 2024 15:39:11 +0200
Subject: [PATCH] ARM: dts: stm32: move dfsdm nodes to generic binding on
 stm32mp157x-ev1

Replace the legacy channel binding by the generic channel binding in
DFSDM nodes stm32mp157x-ev1 device trees.

Change-Id: Ia7adbc3d2d104304fe23e4bbdfa1e5d5a6ce5285
Signed-off-by: Olivier Moysan <olivier.moysan@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/408562
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 arch/arm/boot/dts/st/stm32mp157a-ev1.dts | 56 ++++++++++++++++--------
 arch/arm/boot/dts/st/stm32mp157c-ev1.dts | 56 ++++++++++++++++--------
 arch/arm/boot/dts/st/stm32mp157d-ev1.dts | 56 ++++++++++++++++--------
 arch/arm/boot/dts/st/stm32mp157f-ev1.dts | 56 ++++++++++++++++--------
 4 files changed, 152 insertions(+), 72 deletions(-)

diff --git a/arch/arm/boot/dts/st/stm32mp157a-ev1.dts b/arch/arm/boot/dts/st/stm32mp157a-ev1.dts
index f70d077207a5..2a4d941fc450 100644
--- a/arch/arm/boot/dts/st/stm32mp157a-ev1.dts
+++ b/arch/arm/boot/dts/st/stm32mp157a-ev1.dts
@@ -223,13 +223,18 @@ &dfsdm {
 
 	dfsdm0: filter@0 {
 		compatible = "st,stm32-dfsdm-dmic";
-		st,adc-channels = <3>;
-		st,adc-channel-names = "dmic_u1";
-		st,adc-channel-types = "SPI_R";
-		st,adc-channel-clk-src = "CLKOUT";
 		st,filter-order = <3>;
+		#address-cells = <1>;
+		#size-cells = <0>;
 		status = "okay";
 
+		channel@3 {
+			reg = <3>;
+			label = "dmic_u1";
+			st,adc-channel-type = "SPI_R";
+			st,adc-channel-clk-src = "CLKOUT";
+		};
+
 		asoc_pdm0: dfsdm-dai {
 			compatible = "st,stm32h7-dfsdm-dai";
 			#sound-dai-cells = <0>;
@@ -246,14 +251,19 @@ dfsdm_endpoint0: endpoint {
 
 	dfsdm1: filter@1 {
 		compatible = "st,stm32-dfsdm-dmic";
-		st,adc-channels = <0>;
-		st,adc-channel-names = "dmic_u2";
-		st,adc-channel-types = "SPI_F";
-		st,adc-channel-clk-src = "CLKOUT";
 		st,filter-order = <3>;
-		st,adc-alt-channel = <1>;
+		#address-cells = <1>;
+		#size-cells = <0>;
 		status = "okay";
 
+		channel@0 {
+			reg = <0>;
+			label = "dmic_u2";
+			st,adc-channel-type = "SPI_F";
+			st,adc-channel-clk-src = "CLKOUT";
+			st,adc-alt-channel;
+		};
+
 		asoc_pdm1: dfsdm-dai {
 			compatible = "st,stm32h7-dfsdm-dai";
 			#sound-dai-cells = <0>;
@@ -270,14 +280,19 @@ dfsdm_endpoint1: endpoint {
 
 	dfsdm2: filter@2 {
 		compatible = "st,stm32-dfsdm-dmic";
-		st,adc-channels = <2>;
-		st,adc-channel-names = "dmic_u3";
-		st,adc-channel-types = "SPI_F";
-		st,adc-channel-clk-src = "CLKOUT";
-		st,adc-alt-channel = <1>;
 		st,filter-order = <3>;
+		#address-cells = <1>;
+		#size-cells = <0>;
 		status = "okay";
 
+		channel@2 {
+			reg = <2>;
+			label = "dmic_u3";
+			st,adc-channel-type = "SPI_F";
+			st,adc-channel-clk-src = "CLKOUT";
+			st,adc-alt-channel;
+		};
+
 		asoc_pdm2: dfsdm-dai {
 			compatible = "st,stm32h7-dfsdm-dai";
 			#sound-dai-cells = <0>;
@@ -294,13 +309,18 @@ dfsdm_endpoint2: endpoint {
 
 	dfsdm3: filter@3 {
 		compatible = "st,stm32-dfsdm-dmic";
-		st,adc-channels = <1>;
-		st,adc-channel-names = "dmic_u4";
-		st,adc-channel-types = "SPI_R";
-		st,adc-channel-clk-src = "CLKOUT";
 		st,filter-order = <3>;
+		#address-cells = <1>;
+		#size-cells = <0>;
 		status = "okay";
 
+		channel@1 {
+			reg = <1>;
+			label = "dmic_u4";
+			st,adc-channel-type = "SPI_R";
+			st,adc-channel-clk-src = "CLKOUT";
+		};
+
 		asoc_pdm3: dfsdm-dai {
 			compatible = "st,stm32h7-dfsdm-dai";
 			#sound-dai-cells = <0>;
diff --git a/arch/arm/boot/dts/st/stm32mp157c-ev1.dts b/arch/arm/boot/dts/st/stm32mp157c-ev1.dts
index dcf64bce048b..23f863ac8fa2 100644
--- a/arch/arm/boot/dts/st/stm32mp157c-ev1.dts
+++ b/arch/arm/boot/dts/st/stm32mp157c-ev1.dts
@@ -224,13 +224,18 @@ &dfsdm {
 
 	dfsdm0: filter@0 {
 		compatible = "st,stm32-dfsdm-dmic";
-		st,adc-channels = <3>;
-		st,adc-channel-names = "dmic_u1";
-		st,adc-channel-types = "SPI_R";
-		st,adc-channel-clk-src = "CLKOUT";
 		st,filter-order = <3>;
+		#address-cells = <1>;
+		#size-cells = <0>;
 		status = "okay";
 
+		channel@3 {
+			reg = <3>;
+			label = "dmic_u1";
+			st,adc-channel-type = "SPI_R";
+			st,adc-channel-clk-src = "CLKOUT";
+		};
+
 		asoc_pdm0: dfsdm-dai {
 			compatible = "st,stm32h7-dfsdm-dai";
 			#sound-dai-cells = <0>;
@@ -247,14 +252,19 @@ dfsdm_endpoint0: endpoint {
 
 	dfsdm1: filter@1 {
 		compatible = "st,stm32-dfsdm-dmic";
-		st,adc-channels = <0>;
-		st,adc-channel-names = "dmic_u2";
-		st,adc-channel-types = "SPI_F";
-		st,adc-channel-clk-src = "CLKOUT";
 		st,filter-order = <3>;
-		st,adc-alt-channel = <1>;
+		#address-cells = <1>;
+		#size-cells = <0>;
 		status = "okay";
 
+		channel@0 {
+			reg = <0>;
+			label = "dmic_u2";
+			st,adc-channel-type = "SPI_F";
+			st,adc-channel-clk-src = "CLKOUT";
+			st,adc-alt-channel;
+		};
+
 		asoc_pdm1: dfsdm-dai {
 			compatible = "st,stm32h7-dfsdm-dai";
 			#sound-dai-cells = <0>;
@@ -271,14 +281,19 @@ dfsdm_endpoint1: endpoint {
 
 	dfsdm2: filter@2 {
 		compatible = "st,stm32-dfsdm-dmic";
-		st,adc-channels = <2>;
-		st,adc-channel-names = "dmic_u3";
-		st,adc-channel-types = "SPI_F";
-		st,adc-channel-clk-src = "CLKOUT";
-		st,adc-alt-channel = <1>;
 		st,filter-order = <3>;
+		#address-cells = <1>;
+		#size-cells = <0>;
 		status = "okay";
 
+		channel@2 {
+			reg = <2>;
+			label = "dmic_u3";
+			st,adc-channel-type = "SPI_F";
+			st,adc-channel-clk-src = "CLKOUT";
+			st,adc-alt-channel;
+		};
+
 		asoc_pdm2: dfsdm-dai {
 			compatible = "st,stm32h7-dfsdm-dai";
 			#sound-dai-cells = <0>;
@@ -295,13 +310,18 @@ dfsdm_endpoint2: endpoint {
 
 	dfsdm3: filter@3 {
 		compatible = "st,stm32-dfsdm-dmic";
-		st,adc-channels = <1>;
-		st,adc-channel-names = "dmic_u4";
-		st,adc-channel-types = "SPI_R";
-		st,adc-channel-clk-src = "CLKOUT";
 		st,filter-order = <3>;
+		#address-cells = <1>;
+		#size-cells = <0>;
 		status = "okay";
 
+		channel@1 {
+			reg = <1>;
+			label = "dmic_u4";
+			st,adc-channel-type = "SPI_R";
+			st,adc-channel-clk-src = "CLKOUT";
+		};
+
 		asoc_pdm3: dfsdm-dai {
 			compatible = "st,stm32h7-dfsdm-dai";
 			#sound-dai-cells = <0>;
diff --git a/arch/arm/boot/dts/st/stm32mp157d-ev1.dts b/arch/arm/boot/dts/st/stm32mp157d-ev1.dts
index 78c790bdfcfd..5a097af86422 100644
--- a/arch/arm/boot/dts/st/stm32mp157d-ev1.dts
+++ b/arch/arm/boot/dts/st/stm32mp157d-ev1.dts
@@ -223,13 +223,18 @@ &dfsdm {
 
 	dfsdm0: filter@0 {
 		compatible = "st,stm32-dfsdm-dmic";
-		st,adc-channels = <3>;
-		st,adc-channel-names = "dmic_u1";
-		st,adc-channel-types = "SPI_R";
-		st,adc-channel-clk-src = "CLKOUT";
 		st,filter-order = <3>;
+		#address-cells = <1>;
+		#size-cells = <0>;
 		status = "okay";
 
+		channel@3 {
+			reg = <3>;
+			label = "dmic_u1";
+			st,adc-channel-type = "SPI_R";
+			st,adc-channel-clk-src = "CLKOUT";
+		};
+
 		asoc_pdm0: dfsdm-dai {
 			compatible = "st,stm32h7-dfsdm-dai";
 			#sound-dai-cells = <0>;
@@ -246,14 +251,19 @@ dfsdm_endpoint0: endpoint {
 
 	dfsdm1: filter@1 {
 		compatible = "st,stm32-dfsdm-dmic";
-		st,adc-channels = <0>;
-		st,adc-channel-names = "dmic_u2";
-		st,adc-channel-types = "SPI_F";
-		st,adc-channel-clk-src = "CLKOUT";
 		st,filter-order = <3>;
-		st,adc-alt-channel = <1>;
+		#address-cells = <1>;
+		#size-cells = <0>;
 		status = "okay";
 
+		channel@0 {
+			reg = <0>;
+			label = "dmic_u2";
+			st,adc-channel-type = "SPI_F";
+			st,adc-channel-clk-src = "CLKOUT";
+			st,adc-alt-channel;
+		};
+
 		asoc_pdm1: dfsdm-dai {
 			compatible = "st,stm32h7-dfsdm-dai";
 			#sound-dai-cells = <0>;
@@ -270,14 +280,19 @@ dfsdm_endpoint1: endpoint {
 
 	dfsdm2: filter@2 {
 		compatible = "st,stm32-dfsdm-dmic";
-		st,adc-channels = <2>;
-		st,adc-channel-names = "dmic_u3";
-		st,adc-channel-types = "SPI_F";
-		st,adc-channel-clk-src = "CLKOUT";
-		st,adc-alt-channel = <1>;
 		st,filter-order = <3>;
+		#address-cells = <1>;
+		#size-cells = <0>;
 		status = "okay";
 
+		channel@2 {
+			reg = <2>;
+			label = "dmic_u3";
+			st,adc-channel-type = "SPI_F";
+			st,adc-channel-clk-src = "CLKOUT";
+			st,adc-alt-channel;
+		};
+
 		asoc_pdm2: dfsdm-dai {
 			compatible = "st,stm32h7-dfsdm-dai";
 			#sound-dai-cells = <0>;
@@ -294,13 +309,18 @@ dfsdm_endpoint2: endpoint {
 
 	dfsdm3: filter@3 {
 		compatible = "st,stm32-dfsdm-dmic";
-		st,adc-channels = <1>;
-		st,adc-channel-names = "dmic_u4";
-		st,adc-channel-types = "SPI_R";
-		st,adc-channel-clk-src = "CLKOUT";
 		st,filter-order = <3>;
+		#address-cells = <1>;
+		#size-cells = <0>;
 		status = "okay";
 
+		channel@1 {
+			reg = <1>;
+			label = "dmic_u4";
+			st,adc-channel-type = "SPI_R";
+			st,adc-channel-clk-src = "CLKOUT";
+		};
+
 		asoc_pdm3: dfsdm-dai {
 			compatible = "st,stm32h7-dfsdm-dai";
 			#sound-dai-cells = <0>;
diff --git a/arch/arm/boot/dts/st/stm32mp157f-ev1.dts b/arch/arm/boot/dts/st/stm32mp157f-ev1.dts
index 6677d518cde3..f21c875a1531 100644
--- a/arch/arm/boot/dts/st/stm32mp157f-ev1.dts
+++ b/arch/arm/boot/dts/st/stm32mp157f-ev1.dts
@@ -225,13 +225,18 @@ &dfsdm {
 
 	dfsdm0: filter@0 {
 		compatible = "st,stm32-dfsdm-dmic";
-		st,adc-channels = <3>;
-		st,adc-channel-names = "dmic_u1";
-		st,adc-channel-types = "SPI_R";
-		st,adc-channel-clk-src = "CLKOUT";
 		st,filter-order = <3>;
+		#address-cells = <1>;
+		#size-cells = <0>;
 		status = "okay";
 
+		channel@3 {
+			reg = <3>;
+			label = "dmic_u1";
+			st,adc-channel-type = "SPI_R";
+			st,adc-channel-clk-src = "CLKOUT";
+		};
+
 		asoc_pdm0: dfsdm-dai {
 			compatible = "st,stm32h7-dfsdm-dai";
 			#sound-dai-cells = <0>;
@@ -248,14 +253,19 @@ dfsdm_endpoint0: endpoint {
 
 	dfsdm1: filter@1 {
 		compatible = "st,stm32-dfsdm-dmic";
-		st,adc-channels = <0>;
-		st,adc-channel-names = "dmic_u2";
-		st,adc-channel-types = "SPI_F";
-		st,adc-channel-clk-src = "CLKOUT";
 		st,filter-order = <3>;
-		st,adc-alt-channel = <1>;
+		#address-cells = <1>;
+		#size-cells = <0>;
 		status = "okay";
 
+		channel@0 {
+			reg = <0>;
+			label = "dmic_u2";
+			st,adc-channel-type = "SPI_F";
+			st,adc-channel-clk-src = "CLKOUT";
+			st,adc-alt-channel;
+		};
+
 		asoc_pdm1: dfsdm-dai {
 			compatible = "st,stm32h7-dfsdm-dai";
 			#sound-dai-cells = <0>;
@@ -272,14 +282,19 @@ dfsdm_endpoint1: endpoint {
 
 	dfsdm2: filter@2 {
 		compatible = "st,stm32-dfsdm-dmic";
-		st,adc-channels = <2>;
-		st,adc-channel-names = "dmic_u3";
-		st,adc-channel-types = "SPI_F";
-		st,adc-channel-clk-src = "CLKOUT";
-		st,adc-alt-channel = <1>;
 		st,filter-order = <3>;
+		#address-cells = <1>;
+		#size-cells = <0>;
 		status = "okay";
 
+		channel@2 {
+			reg = <2>;
+			label = "dmic_u3";
+			st,adc-channel-type = "SPI_F";
+			st,adc-channel-clk-src = "CLKOUT";
+			st,adc-alt-channel;
+		};
+
 		asoc_pdm2: dfsdm-dai {
 			compatible = "st,stm32h7-dfsdm-dai";
 			#sound-dai-cells = <0>;
@@ -296,13 +311,18 @@ dfsdm_endpoint2: endpoint {
 
 	dfsdm3: filter@3 {
 		compatible = "st,stm32-dfsdm-dmic";
-		st,adc-channels = <1>;
-		st,adc-channel-names = "dmic_u4";
-		st,adc-channel-types = "SPI_R";
-		st,adc-channel-clk-src = "CLKOUT";
 		st,filter-order = <3>;
+		#address-cells = <1>;
+		#size-cells = <0>;
 		status = "okay";
 
+		channel@1 {
+			reg = <1>;
+			label = "dmic_u4";
+			st,adc-channel-type = "SPI_R";
+			st,adc-channel-clk-src = "CLKOUT";
+		};
+
 		asoc_pdm3: dfsdm-dai {
 			compatible = "st,stm32h7-dfsdm-dai";
 			#sound-dai-cells = <0>;
-- 
2.50.1

